jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildResult.java

```diff
@@ -16,12 +16,47 @@
 
 package com.google.cloud.tools.jib.builder.steps;
 
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
 import com.google.cloud.tools.jib.image.DescriptorDigest;
+import com.google.cloud.tools.jib.image.Image;
+import com.google.cloud.tools.jib.image.Layer;
+import com.google.cloud.tools.jib.image.json.BuildableManifestTemplate;
+import com.google.cloud.tools.jib.image.json.ImageToJsonTranslator;
+import com.google.cloud.tools.jib.json.JsonTemplateMapper;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
 import java.util.Objects;
 
 /** Used to record the results of a build. */
 public class BuildResult {
 
+  /**
+   * Gets a {@link BuildResult} from an {@link Image}.
+   *
+   * @param image the image
+   * @param targetFormat the target format of the image
+   * @return a new {@link BuildResult} with the image's digest and id
+   * @throws IOException if writing the digest or container configuration fails
+   */
+  static BuildResult fromImage(
```

Comment:
Let's add a simple unit test now that this is separated out nicely into its own method.